src/main/api/publication.ts

```diff
@@ -179,40 +178,22 @@ export class PublicationApi implements IPublicationApi {
                 this.dispatchToastRequest(ToastType.DownloadFailed, `[${error}]`);
                 throw new Error(`importOpdsPublication ${error}`);
             }
-            titleView = publication.title;
             this.sendDownloadSuccess(url);
             returnView = this.publicationViewConverter.convertDocumentToView(publication);
         }
-
-        // dispatch notification to user with redux
-        this.dispatchToastRequest(ToastType.DownloadComplete,
```

Comment:
I see you've removed some `dispatchToastRequest`, but there is another one above ( `this.dispatchToastRequest(ToastType.DownloadFailed, `[${error}]`);`).
Is that an oversight, or is this intentional?

Reply:
yes, it's intentional to catch only download error. It can be move to catalog service also. And handle the toast dispatch only on service

Reply:
I think consistency is important here, so that all `dispatchToastRequest` calls are made from the same logical "block" in the app's architecture.

Reply:
i don't know how to do this with the current nesting on our base code with `api` and `service`.